git_data_model.sql

```diff
@@ -1,160 +1,337 @@
+VALUES ('abcd', 1,'message1',1);
+
+INSERT INTO commits (hash, parent_commit ,user_id,messages,repo_id)
+VALUES ('efgh','abcd', 1,'message2',1);
+
+INSERT INTO commits (hash,parent_commit, user_id,messages,repo_id)
+VALUES ('ijkl','efgh', 1,'message3',1);
 
-INSERT INTO commits (commit_hash, parent_commit ,author,messages,dates,repo_id)
-VALUES ('efgh','abcd', 1,'message2', '2022-02-01',1);
+INSERT INTO commits (hash,parent_commit, user_id,messages,repo_id)
+VALUES ('mnop','ijkl', 1,'message4',1);
 
-INSERT INTO commits (commit_hash,parent_commit, author,messages,dates,repo_id)
-VALUES ('ijkl','efgh', 1,'message3', '2022-03-01',1);
+INSERT INTO commits (hash,parent_commit, user_id,messages,repo_id)
+VALUES ('pqrs','mnop', 1,'message5',1);
 
+INSERT INTO commits (hash, user_id,messages,repo_id)
+VALUES ('aaa1', 2,'message5',2);
 
 SELECT * FROM commits;
 
 
 CREATE TABLE branches(
-  branch_id BIGSERIAL PRIMARY KEY NOT NULL,
-  branch_name VARCHAR (255),
-  head_pointer VARCHAR(255) ,
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  name VARCHAR (255) NOT NULL,
+  head_pointer VARCHAR(255) NOT NULL,
   parent_branch BIGINT REFERENCES branches,
-  merge_status VARCHAR (255),
-  dates date NOT NULL,
-  repo_id BIGINT,
-  FOREIGN KEY (head_pointer) REFERENCES commits(commit_hash),
-  FOREIGN KEY (repo_id) REFERENCES repo(repo_id),
-  FOREIGN KEY (parent_branch) REFERENCES branches(branch_id),
-  UNIQUE (branch_id)
+  merge_status VARCHAR (255) ,
+  created_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  updated_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  repo_id BIGINT NOT NULL,
+  FOREIGN KEY (head_pointer) REFERENCES commits(hash),
+  FOREIGN KEY (repo_id) REFERENCES repos(id),
+  FOREIGN KEY (parent_branch) REFERENCES branches(id)
 );
 
+CREATE INDEX index_branches_repo ON branches(repo_id);
 
-INSERT INTO branches (branch_name,head_pointer,merge_status,dates,repo_id)
-VALUES ('master','abcd', 'merged','2022-02-01',1);
 
-INSERT INTO branches (branch_name,head_pointer,parent_branch,merge_status,dates,repo_id)
-VALUES ('branch2','ijkl', 1,'unmerged', '2022-01-01',1);
+INSERT INTO branches (name,head_pointer,merge_status,repo_id)
+VALUES ('master','abcd', 'merged',1);
+
+INSERT INTO branches (name,head_pointer,parent_branch,merge_status,repo_id)
+VALUES ('branch2','ijkl', 1,'unmerged',1);
 
 
 SELECT * FROM branches;
 
 
 CREATE TABLE issues (
-  issue_id BIGSERIAL PRIMARY KEY NOT NULL,
-  title VARCHAR (255),
-  author BIGINT ,
-  statuses TEXT,
-  dates DATE,
-  comments TEXT,
-  repo_id BIGINT,
-  UNIQUE (issue_id),
-  FOREIGN KEY (repo_id) REFERENCES repo(repo_id),
-  FOREIGN KEY (author) REFERENCES users(user_id)
+  id BIGSERIAL PRIMARY KEY NOT NULL,
+  title VARCHAR (255) NOT NULL,
+  user_id BIGINT NOT NULL,
+  created_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  updated_at TIMESTAMP WITHOUT TIME ZONE DEFAULT(NOW() AT TIME ZONE 'UTC'),
+  status TEXT NOT NULL,
```

Comment:
varchar is not suitable for this?

Reply:
yes sir varchar is more suitable . will change it accordingly.